### DISCUSSION TITLE
can the openEMS support planewave source?

### DISCUSSION THREAD

**Discussion by 458887866** [thliebig/openEMS-Project#263] [Q&A Usage] [answered]
can the openEMS support planewave source? If not, how can I revise it to add the planewave source.

**Comment by gadiLhv** [comment]
You can do it with a large waveguide excitation. Place it on the relevant face, and write an appropriate function. 
In matlab: AddExcitation
In python: CSX.AddExcitation
If you need help with that, let me know.

**Reply by thliebig** [reply]
No that is not how a plane wave works...

**Reply by gadiLhv** [reply]
Ugh, you're right.

My original though was that with a large enough aperture, I could, for example use a field distribution. For example, for  propagation in the $\hat{z}$ diretion, and sampled in $\hat{z} = 0$:
```math
\mathbf{E}_{apperture} = \left({E_{0_x}\hat{x} + E_{0_y}\hat{y}}\right)\cos\left({\omega t}\right)
```
However, the moment I need to synthesize something like a field propagating in a direction such as $\hat{k}=k_y\hat{x} + k_z\hat{z}$, it can't be done, as the field will have to take the form of 
```math
\mathbf{E}_{apperture} = \left({E_{0_x}\hat{x} + E_{0_y}\hat{y}}\right)\cos\left({\omega t - k_y y}\right)
```
and the $\cos\left({\omega t - k_y y}\right)$ bit cannot be done.

**Comment by thliebig** [comment, marked as answer]
openEMS has a plane wave source. Have a look at the radar cross section tutorials...

**Reply by 458887866** [reply]
Thanks, I will check this solution.
